paramiko/client.py

```diff
@@ -102,7 +102,12 @@ def load_system_host_keys(self, filename=None):
             except IOError:
                 pass
             return
-        self._system_host_keys.load(filename)
+
+        with open(filename, 'r') as fp:
+            self._system_host_keys.load_fp(fp)
```

Comment:
Since `HostKeys.load()` does exactly this ("with open() ...") you might as well let it, rather than duplicate that (admittedly tiny amount of) code here. Same argument applies to `load_host_keys()` below.